.github/workflows/publish-docs.yml

```diff
@@ -81,6 +81,16 @@ jobs:
           path: site
           if-no-files-found: error
 
+      - name: Cypress run
+        id: cypress_run
+        uses: cypress-io/github-action@f5bea478b80a774068179089247f13d60abc5010 # tag=v2.11.8
+        with:
+          spec: cypress/integration/sample.spec.ts
+          build: make
+          start: mkdocs serve
+          wait-on: 'http://127.0.0.1:8000/'
+        timeout-minutes: 3
+
       - name: Publish docs
-        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main' && steps.cypress_run.conclusion == 'success'
```

Comment:
i don't think we need this change as the workflow will stop on failed actions by default.